### PR TITLE
docs(KAR-26): add initial dev resources doc

### PR DESCRIPTION
**Pull request by beerparty**
This is just an initial draft of the document. I expect a lot of additional material to be added over time.

### CODE REVIEW

**Review by maruiz93 — APPROVED** [review]

File: DEVELOPER_RESOURCES.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+### Go Programming Language
+Suggested resources for working with the Go programming language.
+
+* [Exercism learning track for Go](https://exercism.org/tracks/go)
+* [Golang By Example - Advanced Tutorial](https://golangbyexample.com/golang-comprehensive-tutorial/)
+* [Ardan Labs Ultimate Go Tour](https://tour.ardanlabs.com/tour/eng/list)
+* [Dave Cheney's Practical Go: Real Worl Advice Guide](https://dave.cheney.net/practical-go/presentations/qcon-china.html)
+
+### Add-ons
+Add-ons the KubeArchive uses.
```
Comment (rh-hemartin) [inline DEVELOPER_RESOURCES.md:34]: I would call these "tools" or "related software" rather than "add-ons".

File: DEVELOPER_RESOURCES.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+### Red Hat OpenShift
+Red Hat OpenShift provides enterprise-ready enhancements to Kubernetes.
+This is a list of suggested Red Hat training courses for OpenShift developers.
+
+* [DO188 - Red Hat OpenShift Development I: Introduction to Containers with Podman](https://role.rhu.redhat.com/rol-rhu/app/catalog?q=DO188)
+* [DO288 - Red Hat OpenShift Developer II: Building and Deploying Cloud-native Applications](https://role.rhu.redhat.com/rol-rhu/app/catalog?q=DO288)
+* [DO244 - Developing Applications with Red Hat OpenShift Serverless and Knative](https://role.rhu.redhat.com/rol-rhu/app/catalog?q=DO244)
+* [DO328 - Building Resilient Microservices with Istio and Red Hat OpenShift Service Mesh](https://role.rhu.redhat.com/rol-rhu/app/catalog?q=DO328)
+* [DO378 - Red Hat Cloud-native Microservices Development with Quarkus](https://role.rhu.redhat.com/rol-rhu/app/catalog?q=DO378)
+* [AD482 - Developing Event-driven Applications with Apache Kafka and Red Hat AMQ Streams](https://role.rhu.redhat.com/rol-rhu/app/catalog?q=AD482)
```
Comment (rh-hemartin) [inline DEVELOPER_RESOURCES.md:63]: I would rather just link to the software pages. Linking here paid courses and register-available books from Red Hat does not feel right to me, even if we are working for Red Hat. I think we should aim to maintain this project Red Hat agnostic.

```suggestion
* [Kubernetes - open-source container orchestration](https://kubernetes.io/)
* [OpenTelemetry](https://opentelemetry.io/)

```

**Review by rh-hemartin — CHANGES_REQUESTED** [review]

File: DEVELOPER_RESOURCES.md (hunk 1 above)
Comment (beerparty) [reply DEVELOPER_RESOURCES.md]: Renamed to "Tools".

**Review by rh-hemartin — APPROVED** [review]
Looks good to me now, thanks for the changes. You have three commits here, remember to squash them as they represent one single change.

**Pull request merged**